embedded/document/engine_test.go

```diff
@@ -337,9 +336,12 @@ func TestQueryDocument(t *testing.T) {
 			},
 		}
 
-		doc, err := engine.GetDocuments(context.Background(), collectionName, expressions, 1, 20)
+		reader, err := engine.GetDocuments(ctx, collectionName, expressions, 0)
 		require.NoError(t, err)
-		require.Equal(t, 10, len(doc))
+		defer reader.Close()
+		docs, err := reader.Read(ctx, 20)
+		require.ErrorIs(t, err, sql.ErrNoMoreRows)
```

Comment:
`sql.ErrNoMoreRows` should be catch and `ErrNoMoreDocuments` may be returned instead

Reply:
ack